cinema/views.py

```diff
@@ -45,9 +84,32 @@ def get_serializer_class(self):
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+    queryset = (
+        MovieSession.objects.all()
+        .select_related("movie", "cinema_hall")
+        # .annotate(
+        #     tickets_available=F("cinema_hall__rows")
+        #     * F("cinema_hall__seats_in_row")
+        #     - Count("tickets")
+        # )
+    )
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        date = self.request.query_params.get("date")
+        movie_id_str = self.request.query_params.get("movie")
+
+        queryset = self.queryset
+
+        if date:
+            date = datetime.strptime(date, "%Y-%m-%d").date()
+            queryset = queryset.filter(show_time__date=date)
```

Comment:
In ma case such code works:
 ```
       if date:
            queryset = queryset.filter(show_time__date=date)
```
and you do not need one more raw

Reply:
In my case it unfortunately doesn't work